src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -350,7 +354,11 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             }
 
             info.AddValue(VersionName, _version);
-            info.AddValue(ComparerName, _comparer ?? EqualityComparer<TKey>.Default, typeof(IEqualityComparer<TKey>));
+
+            IEqualityComparer<TKey> comparer = _comparer;
+            // Use public randomized comparer for serialization
+            HashHelpers.TrySwitchRandomizedHashCodeGenerator(ref comparer);
```

Comment:
Why not set the serialization type inside the new Comparer to mask it for serialization instead of manually switching back? I.e. https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/shared/System/StringComparer.cs#L383

Reply:
Is this correct? https://github.com/dotnet/coreclr/pull/19436/commits/31ec133c226d6eda197ba7c503c7c432536ecdc3

Reply:
Yep, looks good, thanks.